src/rust/engine/process_execution/src/remote.rs

```diff
@@ -26,12 +26,79 @@ use super::{
 use std;
 use std::cmp::min;
 use workunit_store::{generate_random_64bit_string, get_parent_id, WorkUnit, WorkUnitStore};
+use std::fmt;
 
 // Environment variable which is exclusively used for cache key invalidation.
 // This may be not specified in an ExecuteProcessRequest, and may be populated only by the
 // CommandRunner.
 const CACHE_KEY_GEN_VERSION_ENV_VAR_NAME: &str = "PANTS_CACHE_KEY_GEN_VERSION";
 
+
+struct CancelRemoteExecutionToken {
+  // Remote execution process needs to be cancel in the case of Fatal error or timeout or
+  // pants run is cancelled by the user.
+  // CancelRemoteExecutionToken is used to cancel remote execution process by sending
+  // CancelOperationRequest.
+  operations_client: Arc<bazel_protos::operations_grpc::OperationsClient>,
+  cancel_op_req: bazel_protos::operations::CancelOperationRequest,
+  drop_enabled: bool,
+}
+
+impl CancelRemoteExecutionToken {
+
+  fn new(
+    operations_client: Arc<bazel_protos::operations_grpc::OperationsClient>,
+    operation_name: ::std::string::String,
+  ) -> CancelRemoteExecutionToken {
+    let mut cancel_op_req = bazel_protos::operations::CancelOperationRequest::new();
+    cancel_op_req.set_name(operation_name);
+    CancelRemoteExecutionToken {
+      operations_client,
+      cancel_op_req,
+      drop_enabled: true,
+    }
+  }
+
+  fn dismiss(&mut self) {
+    self.drop_enabled = false;
+  }
+}
+
+impl Drop for CancelRemoteExecutionToken {
+  fn drop(&mut self) {
+    if self.drop_enabled {
+      let operation_name = self.cancel_op_req.name.clone();
+      match self.operations_client.cancel_operation_async(&self.cancel_op_req) {
+        Ok(receiver) => {
+          tokio::spawn(receiver.then(move |res| {
```

Comment:
To have these log properly, this should probably use `Executor::spawn_and_ignore` from the `Core`'s `executor` field